Sources/Client/Client/Client+Events.swift

```diff
@@ -0,0 +1,133 @@
+//
+//  Client+Events.swift
+//  StreamChatClient
+//
+//  Created by Bahadir Oncel on 30.03.2020.
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// Reference for the subscription initiated. Call `cancel()` to end subscription.
+public protocol Cancellable {
+    /// Cancel the underlying subscription for this object.
+    func cancel()
+}
+
+/// Reference for the subscription initiated. Call `cancel()` to end the subscription.
+/// Alternatively, when this object is deallocated, it calls `cancel()` on itself automatically.
+public protocol AutoCancellable: Cancellable {}
+
+struct Subscription: Cancellable {
```

Comment:
As `SubscriptionBag` this too should be public.

Reply:
We don't want users creating `Subscription`s but we want them to be able to create `SubscriptionBag`, they're 2 different things.